src/DevChatter.Bot.Core/Games/DealNoDeal/DealNoDealGame.cs

```diff
@@ -0,0 +1,342 @@
+
+                GameState = DealNoDealGameState.ChosingStartingBoxes;
+                SetActionForGameState(DealNoDealGameState.ChosingStartingBoxes);
+            }
+            else
+            {
+                _chatClient.SendMessage($"Not enough tokens to start a game, {chatUser}");
+            }
+
+        }
+
+        public void SetActionForGameState(DealNoDealGameState gameState)
+        {
+            if (gameState == DealNoDealGameState.GameNotRunning)
+            {
+                _automatedActionSystem.RemoveAction(_actionBeingPerformed);
+                return;
+            }
+            //Remove all Game-actions and start a new one
+            _automatedActionSystem.RemoveAction(_actionBeingPerformed);
+            _actionBeingPerformed = _automatedActionFactory.GetIntervalAction(gameState);
+            _automatedActionSystem.AddAction(_actionBeingPerformed);
+        }
+
+        public void QuitGame(IChatClient chatClient)
+        {
+            IsRunning = false;
+            MainPlayer = null;
+            DealOffer = 0;
+            GameState = DealNoDealGameState.GameNotRunning;
+            SetActionForGameState(DealNoDealGameState.GameNotRunning);
+        }
+
+        private void SendGameAlreadyStartedMessage(IChatClient chatClient, ChatUser chatUser)
+        {
+            chatClient.SendMessage(
+                $"There is already a {nameof(DealNoDealGame)} running by {MainPlayer.DisplayName}, {chatUser.DisplayName}. Wait for the game to finish!");
+        }
+
+        private void SendGameNotStartedMessage(IChatClient chatClient, ChatUser chatUser)
+        {
+            chatClient.SendMessage($"There's no {nameof(DealNoDealGame)} running, {chatUser.DisplayName}.");
+        }
+
+        private bool ChargeTokensForStartingAGame(IChatClient chatClient, ChatUser chatUser)
+        {
+            chatClient.SendMessage($"{TOKENS_TO_PLAY} tokens are being charged for starting a game.");
+            return _currencyGenerator.RemoveCurrencyFrom(chatUser.DisplayName, TOKENS_TO_PLAY);
+        }
+
+        public void UpdateGameState()
+        {
+            if (CheckIfGameIsWon())
+            {
+                return;
+            }
+
+            //Note: if the box count is between 2 and totalcount - 2
+            int totalPlayableBoxesCount = InitialPrices.Count - StartingBoxes.Count;
+            bool shouldOfferDeal = (BoxesWithOwners.Count < totalPlayableBoxesCount - 2) &&
+                                   (BoxesWithOwners.Count != 1) &&
+                                   (BoxesWithOwners.Count % 2 == 0);
+            if (shouldOfferDeal)
+            {
+                DealOffer = GetADeal();
+                GameState = DealNoDealGameState.MakingADeal;
+                SetActionForGameState(DealNoDealGameState.MakingADeal);
+            }
+            else
+            {
+                //Restart timer
```

Comment:
This comment is confusing, what does the code below have to do with restarting the timer?